framework/util/platform.h

```diff
@@ -343,30 +356,50 @@ inline int32_t FileVprintf(FILE* stream, const char* format, va_list vlist)
 
 inline int32_t LocalTime(tm* local_time, const time_t* timer)
 {
+#if defined(__ANDROID__) || defined(__USE_POSIX)
+    tm* result = localtime_r(timer, local_time);
+    return (result != nullptr) ? 0 : errno;
+#else
+    if (local_time == nullptr)
+    {
+        return EINVAL;
+    }
+
     tm* result = localtime(timer);
     if (result != nullptr)
     {
-        memcpy(local_time, result, sizeof(tm));
+        (*local_time) = (*result);
         return 0;
     }
     else
     {
         return errno;
     }
+#endif
 }
 
 inline int32_t GMTime(tm* gm_time, const time_t* timer)
```

Comment:
Remove this?

Reply:
This is actually used when creating the time string for the timestamped filenames, where there is an option to use either the local timezone or GMT.